demodel/charbuffer.go

```diff
@@ -27,36 +27,46 @@ func (c *CharBuffer) ResetTagline() error {
 	return nil
 }
 
-func getSnarfSaveDir() string {
+// getHomeDir always returns some sort of home directory
+func getHomeDir() string {
 	u, err := user.Current()
 	if err != nil {
-		return ""
+		return os.Getenv("HOME")
 	}
-	return u.HomeDir + "/.de/snarf/"
+	return u.HomeDir
+}
+
+// ConfigHome returns the directory where configuration files are written
+func ConfigHome() string {
+	return getHomeDir() + "/.de"
+}
+
+// DataHome returns the directory where data files may be written
+func DataHome() string {
+	return getHomeDir() + "/.de"
+}
+
+func getSnarfSaveDir() string {
+	return DataHome() + "/snarf/"
+}
```

Comment:
These three functions should probably either all include a / at the end, or all not include a / at the end, in order to make it easier to maintain in the long term